packages/web/src/generic.ts

```diff
@@ -9,28 +16,26 @@ export const inject = (
 ): void => {
   if (!isBrowser()) return;
 
-  const mode = getMode(props.mode);
-
-  window.vam = mode;
+  const mode = detectMode(props.mode);
+  setMode(mode);
```

Comment:
Could we please combine those things? What about `setMode(props.mode)` only as we use `isDevelopment` and `isProduction`